nest-forms-backend/src/nases/utils-services/tokens.nases.service.ts

```diff
@@ -409,48 +442,82 @@ export default class NasesUtilsService {
     let subject: string = form.id
     let mimeType = 'application/x-eform-xml'
     let encoding = 'XML'
-    let attachments = ''
+    let attachments: NasesAttachementXmlObject[] = []
 
     if (isSlovenskoSkTaxFormDefinition(formDefinition)) {
       subject = 'Podávanie daňového priznanie k dani z nehnuteľností' // TODO fix in formDefinition, quickfix here formDefinition.messageSubjectDefault
       mimeType = 'application/vnd.etsi.asic-e+zip'
       encoding = 'Base64'
       attachments = await this.createAttachmentsIfExists(form, formDefinition)
     }
-    envelope = `
-            <?xml version="1.0" encoding="utf-8"?>
-            <SKTalkMessage xmlns="http://gov.sk/SKTalkMessage" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
-              <EnvelopeVersion>3.0</EnvelopeVersion>
-              <Header>
-                <MessageInfo>
-                  <Class>EGOV_APPLICATION</Class>
-                  <PospID>${pospID}</PospID>
-                  <PospVersion>${pospVersion}</PospVersion>
-                  <MessageID>${form.id}</MessageID>
-                  <CorrelationID>${correlationId}</CorrelationID>
-                </MessageInfo>
-              </Header>
-              <Body>
-                <MessageContainer xmlns="http://schemas.gov.sk/core/MessageContainer/1.0">
-                  <MessageId>${form.id}</MessageId>
-                  <SenderId>${senderId}</SenderId>
-                  <RecipientId>${
-                    process.env.NASES_RECIPIENT_URI ?? ''
-                  }</RecipientId>
-                  <MessageType>${pospID}</MessageType>
-                  <MessageSubject>${subject}</MessageSubject>
-                  <Object Id="${form.id}" IsSigned="${
-                    isSigned ? 'true' : 'false'
-                  }" Name="${title}" Description="" Class="FORM" MimeType="${mimeType}" Encoding="${encoding}">${message}</Object>
-    ${attachments}
-                </MessageContainer>
-              </Body>
-            </SKTalkMessage>
-      `
+
+    const attachmentBase = {
+      $: {
+        Id: form.id,
+        IsSigned: isSigned ? 'true' : 'false',
+        Name: title,
+        Description: '',
+        Class: 'FORM',
+        MimeType: mimeType,
+        Encoding: encoding,
+      },
+    }
+
+    if (typeof message === 'string') {
+      attachments.push({
+        ...attachmentBase,
+        _: message,
+      })
+    } else if (typeof message === 'object') {
+      attachments.push({
+        ...attachmentBase,
+        ...message,
+      })
+    }
```

Comment:
I don't know if I understand this correctly, but:
1. The message is passed in the attachments (Inner text in <Object>)? For this case, wouldn't it be more understandable to rename attachments to message, since it contains both attachments and the message?
2. Was this tested both with messages in string and object format?

Reply:
1. I thought of it as a "Attached is a message...", but I can change it
2. Yes, but I realized I did not test another part of the code - will fix